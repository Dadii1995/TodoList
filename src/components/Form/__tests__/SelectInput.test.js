import renderer from 'react-test-renderer'
import React from 'react'
import { Field, Formik } from 'formik'
import SelectInput from '../SelectInput'

const getWrapper = initialState => {
  const wrapper = renderer.create(
    <Formik initialValues={initialState.initialValues}>
      <Field
        component={SelectInput}
        label={initialState.label}
        name={initialState.name}
        options={initialState.options}
      />
    </Formik>,
  )
  const root = wrapper.root
  return { root, wrapper }
}
describe('<SelectInput/>', () => {
  test('renders select without options', () => {
    const initialState = {
      name: 'selectInput',
      label: 'Select:',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('renders select with initial value', () => {
    const initialState = {
      name: 'selectInput',
      label: 'Select:',
      options: [
        {
          value: 'Val1',
          label: 'Lab1',
        },
        {
          value: 'Val2',
          label: 'Lab2',
        },
        {
          value: 'Val3',
          label: 'Lab3',
        },
        {
          value: 'Val4',
          label: 'Lab4',
        },
        {
          value: 'Val5',
          label: 'Lab5',
        },
      ],
      initialValues: {
        selectInput: 'Val2',
      },
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('renders select without options', () => {
    const initialState = {
      name: 'selectInput',
      label: 'Select:',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
})
