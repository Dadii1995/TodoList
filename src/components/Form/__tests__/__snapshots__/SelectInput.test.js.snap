// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput/> renders select with initial value 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Select:
  </label>
  <select
    aria-label="Select:"
    className="form-control"
    name="selectInput"
    onBlur={[Function]}
    onChange={[Function]}
    value="Val2"
  >
    <option
      value="Val1"
    >
      Lab1
    </option>
    <option
      value="Val2"
    >
      Lab2
    </option>
    <option
      value="Val3"
    >
      Lab3
    </option>
    <option
      value="Val4"
    >
      Lab4
    </option>
    <option
      value="Val5"
    >
      Lab5
    </option>
  </select>
</div>
`;

exports[`<SelectInput/> renders select without options 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Select:
  </label>
  <select
    aria-label="Select:"
    className="form-control"
    name="selectInput"
    onBlur={[Function]}
    onChange={[Function]}
  />
</div>
`;

exports[`<SelectInput/> renders select without options 2`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Select:
  </label>
  <select
    aria-label="Select:"
    className="form-control"
    name="selectInput"
    onBlur={[Function]}
    onChange={[Function]}
  />
</div>
`;
